src/main/java/net/dries007/tfc/world/classic/worldgen/WorldGenPlantTFC.java

```diff
@@ -113,7 +115,12 @@ public boolean generate(World worldIn, Random rand, BlockPos position)
                 for (int i = 0; i < ChunkDataTFC.getRainfall(worldIn, position) / 16; ++i)
                 {
                     BlockPos blockpos = position.add(rand.nextInt(4) - rand.nextInt(4), rand.nextInt(4) - rand.nextInt(4), rand.nextInt(4) - rand.nextInt(4));
-
+                    //Make sure this is a good position to place (eg: not on top of leaf blocks, replaceable blocks)
+                    while (!worldIn.isAirBlock(blockpos) || worldIn.getBlockState(blockpos.down()).getBlock() instanceof BlockLeaves || worldIn.getBlockState(blockpos.down()).getMaterial().isReplaceable())
```

Comment:
This shouldn't be duct-taped on here - it should become part of `BlockPlantTFC` or whichever is used for moss type plants. Ideally there's a flag for if it requires a solid block to support it. May require an additional plant type.